### PR TITLE
Add class 6

### PR DESCRIPTION
**Pull request by EvanDorsky**
Starting with the CSS section. In the Google doc @benkahle expressed concern that this was "too much for one class." According to that we should also cover:
- Development style/grace
  - ~~MVC breakup~~ and file organization
  - separation of concerns
  - ~~function documentation~~
- Session/user management
  - cookies
  - ~~Redis~~
  - ~~security concerns~~
- Error handling
  - programmer vs. operator
  - ~~error vs. exception~~
  - ~~async vs. sync~~

Do any of those topics fit better in other classes?

Crossed out things that weren't covered, also added a lot of stuff (but not to the list above).

### CODE REVIEW

File: classes/class5/README.md [hunk 1]
```diff
@@ -1 +1,124 @@
...
+
+In the image above, the solid border around the yellow box defines the outer edge of the element. The `margin` is space between the box and its surroundings. The `border` is space between the outer edge and the `padding`, which defines the space between the inner edge of the border and the content of the element.
+
+As the image above shows, the `width` property sets the width of the *content*, not the width of the entire element. This means that `padding` and `border` add to the apparent dimensions of the box. We can add the below property to change that:
+
+```css
+box-sizing: border-box;
+```
+
+With this property set, the `width` and `height` properties set the dimensions of the box, and those dimensions are not affected by `padding` or `border`.
```
Comment (benkahle) [inline classes/class5/README.md]: Perhaps useful to mention benefits either way of these models (or maybe something we just mention in class)

File: classes/class5/README.md [hunk 2]
```diff
@@ -1 +1,124 @@
...
+Properties can be named in any order, but in the case of directly conflicting properties, the last one defined will apply. Many property names are fairly self-explanatory, but the [full property table](http://www.w3.org/TR/CSS21/propidx.html) may serve as a useful reference. Or Google.
+
+### Units
+
+Many properties values are lengths, which can be specified in many different ways in CSS. The units below are the most common.
+
+| Unit  | Description |
+|----|----|
+| `px` | Usually a single pixel on the client's screen (a line of width `1px` is guaranteed to be "sharp and visible")
+| `em` | Relative to the font-size of the element (`0.5em` is half of the current font-size)
```
Comment (benkahle) [inline classes/class5/README.md]: Also `rem`

File: classes/class5/README.md [hunk 3]
```diff
@@ -1 +1,124 @@
...
+| `%` | Percent of the parent element along the relevant dimension (horizontally or vertically)
+
+CSS also supports the absolute units `cm`, `mm`, `in`, `pt`, and `pc`, but these are better suited for print than the screen.
+
+### Selectors and Specificity
+
+The most common selector is a `tagname`, a `class` (preceded by `.`), or an `id` (preceded by `#`), though there are many other ways to select elements to style.
```
Comment (benkahle) [inline classes/class5/README.md]: common selectors are

File: classes/class5/README.md [hunk 4]
```diff
@@ -1 +1,124 @@
...
+|----|----
+| `#RRGGBB` | Red, green, and blue are each specified by a two-digit hex number, `00` to `FF`. White is `#FFFFFF` and black is `#000000`.
+| `#RGB` | Same as above but with half the precision.
+| `name` | Certain colors (like red, blue, yellow, etc.) can be specified simply by name. They're mostly hideous but black and white are fine.
+| `rgb(r,g,b)` | Red, green, and blue are specified on a 0-255 scale.
+| `rgba(r,g,b,a)` | Same as above, but alpha (opacity) is specified on a 0-1 scale.
+
+Here's a neat [clock](http://whatcolourisit.scn9a.org) that displays the color specified by the hex code corresponding to the current time.
+
+### Box Model
```
Comment (benkahle) [inline classes/class5/README.md]: Probably useful to discuss inline vs. block vs. inline-block elements and absolute vs relative positioning.

**Comment by benkahle** [comment]
Probably useful to discuss inline vs. block vs. inline-block elements and absolute vs relative positioning.
Also floating and stuff like that.

At this point it seems like it shouldn't be too difficult to get the other topics in this class. We have been moving pretty quickly in class time.

File: classes/class06/README.md [hunk 5]
```diff
@@ -1 +1,266 @@
-#Class 5
+# Class5 - CSS, Development Style and Grace
```
Comment (EvanSimpson) [inline classes/class06/README.md]: Class 6

File: classes/class06/README.md [hunk 6]
```diff
@@ -1 +1,266 @@
...
+
+- **HTML** specifies the _structure_ and _content_ of a webpage.
+- **CSS** specifies the _presentation_ or _appearance_ of elements on a webpage.
+- **JavaScript** lets the developer create _interactive_ elements. It's how the client communicates with the server and the user.
+
+Before CSS, HTML was also used to control the appearance of webpages. This is why websites from the '90s look terrible.
+
+These three technologies embody the principle of _separation of concerns_ — they are each completely responsible for a single aspect of a complex web app.
+
+## App file structure/hierarchy
```
Comment (EvanSimpson) [inline classes/class06/README.md:19]: Jumping from front-end to back-end without explicitly saying it. "Is this some new front-end app thing that is a part of that **JavaScript** bullet point above?"

File: classes/class06/README.md [hunk 7]
```diff
@@ -1 +1,266 @@
...
+
+- **/models** - contains ORM models (mongoose `Schemas`)
+- **/views** - view-templates
+- **/public** - all static content
+  - **/images**
+  - **/stylesheets**
+  - **/javascripts**
+- **/routes** - all express routes, separated by app module/area
```
Comment (EvanSimpson) [inline classes/class06/README.md:30]: Are we going to briefly touch on MVC? If so, clarify that our controllers live in routes. They can be separated, but for simple applications, it's acceptable to write controller logic directly in to route assignment. Mostly relevant to using the breakout router object.

File: classes/class06/README.md [hunk 8]
```diff
@@ -1 +1,266 @@
...
+- **/tests** - for unit-tests
+- **/node_modules** - created by `npm`
+
+## CSS
+
+Open up [Codepen](http://codepen.io/pen/), we're going to learn CSS!
```
Comment (EvanSimpson) [inline classes/class06/README.md:76]: we're going to ~~attemp to~~ learn

Comment (EvanDorsky) [inline classes/class06/README.md:76]: learn css ?

File: classes/class06/README.md [hunk 9]
```diff
@@ -1 +1,266 @@
...
+
+Client-side web development is dominated by three technologies.
+
+- **HTML** specifies the _structure_ and _content_ of a webpage.
+- **CSS** specifies the _presentation_ or _appearance_ of elements on a webpage.
+- **JavaScript** lets the developer create _interactive_ elements. It's how the client communicates with the server and the user.
+
+Before CSS, HTML was also used to control the appearance of webpages. This is why websites from the '90s look terrible.
+
+These three technologies embody the principle of _separation of concerns_ — they are each completely responsible for a single aspect of a complex web app.
```
Comment (benkahle) [inline classes/class06/README.md:17]: Link to wiki article or other source on "separation of concerns"?

File: classes/class06/README.md (hunk 6 above)
Comment (benkahle) [inline classes/class06/README.md:19]: What's the point of this section? It seems a bit disjointed in its current state.

File: classes/class06/README.md (hunk 5 above)
Comment (EvanDorsky) [inline classes/class06/README.md]: Yup

File: classes/class06/README.md (hunk 8 above)
Comment (EvanSimpson) [inline classes/class06/README.md:76]: :+1:

File: classes/class06/sessions/app.js [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(cookieParser());
+app.use(session({
+	secret: 'secret',
+	resave: false,
+	saveUninitialized: true
+}));
+
+app.get('/', function(req, res) {
+	console.dir(req.cookies);
```
Comment (benkahle) [inline classes/class06/sessions/app.js:22]: Reason for dir over log?

**Comment by EvanDorsky** [comment]
The blue squares are delete buttons. I'll probably update the picture later. And maybe take delete buttons off of twotes you can't delete.

Also re: autoscroll I think those are great things for them to discover if they want to do the optional thing with the user list.

File: classes/class06/sessions/app.js (hunk 10 above)
Comment (EvanDorsky) [inline classes/class06/sessions/app.js:22]: Dir formats objects more nicely. Also it's a chance to teach people about it.

File: classes/class06/sessions/app.js [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+	console.dir(req.session);
+	
+	if (req.session.counter)
+		req.session.counter++;
+	else
+		req.session.counter = 1;
+
+	res.send('hello');
```
Comment (benkahle) [inline classes/class06/sessions/app.js]: Could you send the counter itself or "Hello, were glad to see you've returned "+counter+" times!" to make it a bit easier to see without requiring the cookie viewing in the browser?

File: classes/class06/sessions/app.js (hunk 10 above)
Comment (benkahle) [inline classes/class06/sessions/app.js:22]: http://nodejs.org/api/console.html#console_console_log_data

Console.log can take multiple arguments and can use string formatting, but a single argument call to log and dir both use util.inspect() the same way.

**Comment by benkahle** [comment]
I'm calling this good for class at this point. We can fix the homework notes as a bugfix.

**Pull request merged**